src/Foundatio.RabbitMQ/Messaging/RabbitMQMessageBus.cs

```diff
@@ -93,6 +93,15 @@ public RabbitMQMessageBus(Builder<RabbitMQMessageBusOptionsBuilder, RabbitMQMess
             SendMessageToSubscribers(message, _serializer);
         }
 
+        /// <summary>
+        /// Deserialize message's body
+        /// </summary>
+        /// <param name="messageBody">raw body that was received from RabbitMQ</param>
+        /// <returns>deserialized representation of body</returns>
+        protected virtual MessageBusData DeserializeMessage(byte[] messageBody) {
```

Comment:
Perhaps this should be moved up into message bus base.

Reply:
It shouldn't be moved to message bus base because serialization and deserialization depends on implementation of message bus. As I suppose usage of `MessageBusData` takes root deep into the message bus base. Thus, I think that it is not worth changing the message bus base, but it is better to conduct an experiment on one library.